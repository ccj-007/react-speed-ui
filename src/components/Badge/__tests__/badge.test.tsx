import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Badge, Avatar } from '../../index'

describe('test badge component', () => {
  it('basic use should render', () => {
    const { asFragment } = render(
      <>
        <Badge>
          <Avatar shape="square" text='User'></Avatar>
        </Badge>
        <Badge count={0} showZero>
          <Avatar shape="square" text='User'></Avatar>
        </Badge>
        <Badge count={10} dotSize={20} style={{ background: 'green' }}>
          <Avatar shape="square" text='User' ></Avatar>
        </Badge>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('different tag should render', () => {
    const { asFragment } = render(
      <>
        <Badge dot dotSize={5}>
          <Avatar shape="square" text='User'></Avatar>
        </Badge>
        <Badge dot dotSize={10}>
          <Avatar shape="square" text='User'></Avatar>
        </Badge>
        <Badge dot dotSize={15}>
          <Avatar shape="square" text='User'></Avatar>
        </Badge>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})