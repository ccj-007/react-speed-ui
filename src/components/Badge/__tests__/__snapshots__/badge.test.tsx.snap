// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test badge component basic use should render 1`] = `
<DocumentFragment>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 15px; color: red; height: 15px; right: -7.5px; top: -7.5px;"
    >
      1
    </div>
  </div>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 15px; color: red; height: 15px; right: -7.5px; top: -7.5px;"
    >
      0
    </div>
  </div>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 20px; color: red; height: 20px; right: -10px; top: -10px; background: green;"
    >
      10
    </div>
  </div>
</DocumentFragment>
`;

exports[`test badge component different tag should render 1`] = `
<DocumentFragment>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 5px; color: red; height: 5px; right: -2.5px; top: -2.5px;"
    />
  </div>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 10px; color: red; height: 10px; right: -5px; top: -5px;"
    />
  </div>
  <div
    class="speed-avatar speed-badge"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
    <div
      class="speed-badge-numDot"
      style="width: 15px; color: red; height: 15px; right: -7.5px; top: -7.5px;"
    />
  </div>
</DocumentFragment>
`;
